src/NServiceBus.Core.Tests/Sagas/ComponentTests/SagaPersisterTests.cs

```diff
@@ -0,0 +1,208 @@
+        protected Task<TSagaData> GetByIdAndUpdate(Guid sagaId, Action<TSagaData> update, params Type[] availableTypes) => GetByIdAndUpdate<TSaga, TSagaData>(sagaId, update, availableTypes);
+        protected Task<TSagaData> GetByCorrelationPropertyAndUpdate(string correlatedPropertyName, object correlationPropertyData, Action<TSagaData> update) => GetByCorrelationPropertyAndUpdate<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData, update);
+        protected Task<TSagaData> GetByCorrelationPropertyAndComplete(string correlatedPropertyName, object correlationPropertyData) => GetByCorrelationPropertyAndComplete<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData);
+        protected Task<TSagaData> GetByCorrelationProperty(string correlatedPropertyName, object correlationPropertyData) => GetByCorrelationProperty<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData);
+        protected Task<TSagaData> GetById(Guid sagaId, params Type[] availableTypes) => GetById<TSaga, TSagaData>(sagaId, availableTypes);
+    }
+
+    public class SagaPersisterTests
+    {
+        protected PersistenceTestsConfiguration configuration;
+
+        [OneTimeSetUp]
+        public async Task OneTimeSetUp()
+        {
+            configuration = new PersistenceTestsConfiguration();
+            await configuration.Configure();
+        }
+
+        [OneTimeTearDown]
+        public async Task OneTimeTearDown()
+        {
+            await configuration.Cleanup();
+        }
+
+        protected async Task SaveSaga<TSaga, TSagaData>(TSagaData saga, params Type[] availableTypes)
+            where TSaga : Saga<TSagaData>, new()
+            where TSagaData : IContainSagaData, new()
+        {
+            var insertContextBag = configuration.GetContextBagForSagaStorage();
+            using (var insertSession = await configuration.SynchronizedStorage.OpenSession(insertContextBag))
+            {
+                var correlationProperty = SetActiveSagaInstanceForSave(insertContextBag, new TSaga(), saga, availableTypes);
+
+                await configuration.SagaStorage.Save(saga, correlationProperty, insertSession, insertContextBag);
+                await insertSession.CompleteAsync();
+            }
+        }
+
+        protected async Task<TSagaData> GetByIdAndComplete<TSaga, TSagaData>(Guid sagaId, params Type[] availableTypes)
+            where TSaga : Saga<TSagaData>, new()
+            where TSagaData : IContainSagaData, new()
+        {
+            var context = configuration.GetContextBagForSagaStorage();
+            TSagaData sagaData;
+            var persister = configuration.SagaStorage;
+            using (var completeSession = await configuration.SynchronizedStorage.OpenSession(context))
+            {
+                SetActiveSagaInstanceForGet<TSaga, TSagaData>(context, new TSagaData(), availableTypes);
```

Comment:
Yes and no. It is already today possible to pass in any object into the context bag that is being created. The context bag factory method can be overriden. The problem here is the following. As a Taskforce that did the component test we outlined how we want to structure those tests. We said it is going to be a simulation of the core interaction with the persister API. Part of the core interaction is the ActiveSagaInstance is put on the context. We saw that a persister is already relying on and concluded it is the persisters right to do so since the core behaves that way. From a component test perspective the simulation needs to be as close to the core behavior as possible. We then invited people to have a discussion about it. Apparently nobody wanted to take the time. You seem to have a different opinion how the component tests should be structured. That is great and I'm happy to see where this goes. The question is only: Should we do it now as part of this TF? Or later? and how owns the changes? Until we define this it will be a back and forth conversation without no roadmap.